src/Observable.spec.ts

```diff
@@ -45,4 +45,43 @@ describe('Observable', () => {
             }
         )
     })
+
+    it('can filter messages', done => {
+        Observable.of(1, 2, 3).filter(value =>
+            value > 2
+        ).subscribe(value => {
+            expect(value).toBe(3)
+            setTimeout(done)
+        })
+    })
+
+    it('can reduce messages', done => {
+        const pushValues = [1, 2, 3]
+        const expectedValues = [1, 3, 6]
+        let result = []
+        Observable.of(1, 2, 3).reduce((value, summary : number) =>
+            summary + value
+        ).subscribe(value => {
+            result.push(value)
+            expect(value).toBe(expectedValues.shift())
+            if (result.length === 3) {
+                setTimeout(done)
+            }
+        })
+    })
+
+    it('can map messages', done => {
+        const pushValues = [1, 2, 3]
+        const expectedValues = [5, 10, 15]
+        let result = []
+        const subscription = Observable.of(1, 2, 3).map(
```

Comment:
Jak rozumiem to chodzi Ci o taki zapis?
```
const subscription = Observable.of(1, 2, 3)
    .map(...)
```

Czy raczej o drugą część?
```
    v => v * 5
)
    .subscribe(...)
```

Rozumiem że to opcjonalne, ale chciałbym ujednolicić styl pisania z twoim :+1: